### PR TITLE
OCPBUGS-25339: [CARRY] Fix panic issue when annotations map is nil

### PR DESCRIPTION
**Pull request by dinhxuanvu**
In some cases, secrets do not have any annotations and the annotations map are nil. The annotations map needs to be initialized to avoid panic error.

### CODE REVIEW

**Comment by dinhxuanvu** [comment]
@tmshort PTAL. Thanks

**Comment by jianzhangbjz** [comment]
LGTM, test details see: https://issues.redhat.com/browse/OCPBUGS-25339
/label qe-approved
/lgtm

File: staging/operator-lifecycle-manager/pkg/controller/operators/olm/operator.go [hunk 1]
```diff
@@ -2907,19 +2907,29 @@ func (a *Operator) EnsureSecretOwnershipAnnotations() error {
...
+				continue
 			}
+		} else {
+			secret.Annotations = map[string]string{}
+			secret.Annotations[install.OpenShiftComponent] = install.OLMOwnershipAnnotation
+		}
+		logger.Debug("injecting ownership annotations to existing secret")
+		if _, updateErr := a.opClient.UpdateSecret(secret); updateErr != nil {
+			logger.WithError(err).Warn("error adding ownership annotations to existing secret")
+			return err
```
Comment (grokspawn) [inline staging/operator-lifecycle-manager/pkg/controller/operators/olm/operator.go:2932]:
```suggestion
       if secret.Annotations == nil {
          secret.Annotations = map[string]string{}
        }
   	if secret.Annotations[install.OpenShiftComponent] == "" {
				secret.Annotations[install.OpenShiftComponent] = install.OLMOwnershipAnnotation
		} 
		logger.Debug("injecting ownership annotations to existing secret")
		if _, updateErr := a.opClient.UpdateSecret(secret); updateErr != nil {
			logger.WithError(err).Warn("error adding ownership annotations to existing secret")
			return err
```
why not as ^^ ?
(with apologies for the whitespace)

**Comment by tmshort** [comment]
> @tmshort PTAL. Thanks

Sorry, was on vacation.

**Comment by dinhxuanvu** [comment]
> > @tmshort PTAL. Thanks
> 
> Sorry, was on vacation.

All good. Everything was taken off.

**Pull request merged**